sample-apps/delivery-customizations-sample-app/extensions/hide-by-title/src/main.rs

```diff
@@ -0,0 +1,128 @@
+use serde::{Deserialize, Serialize};
+
+mod api;
+use api::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct Configuration {
+    pub delivery_option_name: String,
+}
+
+impl Configuration {
+    fn from_str(value: &str) -> Self {
+        serde_json::from_str(value).expect("Unable to parse configuration value from metafield")
+    }
+}
+
+impl Input {
+    pub fn configuration(&self) -> Configuration {
+        match &self.delivery_customization.metafield {
+            Some(Metafield { value }) => Configuration::from_str(value),
+            None => Configuration::default(),
+        }
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let input: Input = serde_json::from_reader(std::io::BufReader::new(std::io::stdin()))?;
+    let mut out = std::io::stdout();
+    let mut serializer = serde_json::Serializer::new(&mut out);
+    function(input)?.serialize(&mut serializer)?;
+    Ok(())
+}
+
+fn function(input: Input) -> Result<FunctionResult, Box<dyn std::error::Error>> {
+    let delivery_options = delivery_options(&input.cart);
+    let delivery_option_name = input.configuration().delivery_option_name;
+    let operations = delivery_options
+        .iter()
+        .filter_map(|delivery_option| {
+            if &delivery_option_name.as_str() == &delivery_option.title.as_str() {
+                Some(Operation {
+                    hide: Some(HideOperation {
+                        delivery_option_handle: delivery_option.handle.clone(),
+                    }),
+                })
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    Ok(FunctionResult { operations })
+}
+
+fn delivery_options(cart: &Cart) -> Vec<&CartDeliveryOption> {
+    cart.delivery_groups
+        .iter()
+        .flat_map(|delivery_group| &delivery_group.delivery_options)
+        .collect()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    fn input(configuration: Option<Configuration>) -> Input {
+        let input = r#"
+        {
+            "cart": {
+                "deliveryGroups": [
+                    {
+                        "deliveryOptions": [
+                            { "handle": "method-a", "title": "Method A" }
+                        ]
+                    },
+                    {
+                        "deliveryOptions": [
+                            { "handle": "method-b", "title": "Method B" },
+                            { "handle": "method-c", "title": "Method C" }
+                        ]
+                    }
+                ]
+            },
+            "deliveryCustomization": {
+                "metafield": {
+                  "value": "{\"deliveryOptionName\":\"Method A\",\"enabled\":true,\"title\":\"Hide delivery option\",\"operation\":\"Hide\"}"
```

Comment:
Was just 🎩 this but shouldn't this metafield value be the title? Like it would just be `Method A` here

Reply:
Yeah I think `enabled`, `title`, `operation` should all be hardcoded into the page running the mutation and not part of the metafield value